core/src/utilities/helpers/navigation-helpers.js

```diff
@@ -69,6 +69,25 @@ class NavigationHelpersClass {
     return permissionCheckerFn(nodeToCheckPermissionFor, parentNode, currentContext);
   }
 
+  isWebComponentCompoundPermitted(nodeToCheckPermissionFor) {
+    if (nodeToCheckPermissionFor && nodeToCheckPermissionFor.visibleForFeatureToggles) {
+      const activeFeatureToggles = LuigiFeatureToggles.getActiveFeatureToggleList();
+      for (const ft of nodeToCheckPermissionFor.visibleForFeatureToggles) {
+        if (ft.startsWith('!')) {
+          if (activeFeatureToggles.includes(ft.slice(1))) {
+            return false;
+          }
+        } else {
+          if (!activeFeatureToggles.includes(ft)) {
+            return false;
+          }
+        }
+      }
+    }
```

Comment:
This part is almost repeated as function `isNodeAccessPermitted` https://github.com/SAP/luigi/blob/762cae78085b4040425f64c8a389d0aaabff815b/core/src/utilities/helpers/navigation-helpers.js#L51-L64.

Could you please extract this part out?  Follow the DRY (don't repeat yourself) principle. Thanks.

Reply:
Fixed